internal/kubernetes/informer/get_from_index.go

```diff
@@ -0,0 +1,26 @@
+package informer
+
+import (
+	"context"
+
+	"github.com/planetlabs/draino/internal/kubernetes/utils"
+	cachek "k8s.io/client-go/tools/cache"
+	clientcr "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GetSharedIndexInformer interface {
+	GetSharedIndexInformer(ctx context.Context, obj clientcr.Object) (cachek.SharedIndexInformer, error)
+}
+
+func GetFromIndex[T clientcr.Object](ctx context.Context, i GetSharedIndexInformer, idx, val string, obj T) ([]T, error) {
+	indexInformer, err := i.GetSharedIndexInformer(ctx, obj)
```

Comment:
Since obj is only used for its type, would something like this compile?

```suggestion
func GetFromIndex[T clientcr.Object](ctx context.Context, i GetSharedIndexInformer, idx, val string) ([]T, error) {
	var obj T
	indexInformer, err := i.GetSharedIndexInformer(ctx, &obj)
```

Called like this

```go
GetFromIndex[policyv1.PodDisruptionBudget](...)
```

Reply:
I've tried that, but unfortunately the GetSharedIndexInformer will fail with 
```
expected pointer, but got nil
```

Reply:
Oh, right, Pod doesn't implement Object, *Pod does, but we can't use it as T because we need to create a Pod struct.

So we need to use this pattern: https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#pointer-method-example

Simplified application: https://go.dev/play/p/Nlje9NsVLda

Reply:
https://github.com/DataDog/draino/pull/98/commits/b3a9f802561328f2290105d70c9c8238032414fc